app.rb

```diff
@@ -0,0 +1,202 @@
+require_relative 'books'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'person'
+require_relative 'rental'
+
+class App
+  def initialize
+    @book = []
+    @rentals = []
+    @people = []
+  end
+
+  def options
+    puts('Welcome to school library App!')
+    puts('Please choose an option by entering a a number:')
+    puts('1 - List all books')
+    puts('2 - List all people')
+    puts('3 - Create a person')
+    puts('4 - Create a book')
+    puts('5 - Create a rental')
+    puts('6 - List all rentals for a given person id')
+    puts('7 - Exit')
+    puts('')
+  end
+
+  def display
+    options
+  end
+
+  def run_choise(input)
+    case input
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_people
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_of_rentals
+    else
+      exit
+    end
+  end
+
+  def run
+    options
+    puts 'Enter number'
+    input = gets.chomp
+    run_choise(input)
+  end
+
+  # list book
+  def list_books
+    puts 'The Library is empty ,There is no books added yet!' if @book.length.zero?
+
+    @book.each do |book|
+      puts "#{@book.index(book)}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    puts ''
+    run
+  end
+
+  # list people
+  def list_people
+    puts 'There is person in the list!' if @book.length.zero?
+
+    @people.each do |person|
+      puts "Name: #{person.name}  Age: #{person.age}"
+    end
+
+    puts ''
+    run
+  end
+
+  #   Create person
+  def create_people
+    puts 'Do you want to create a student (1) or a teacher (2) ? [Input the number]:'
+    input = gets.chomp.to_i
+    case input
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Please Enter 1/2 please'
+      create_people
+    end
+  end
+
+  # create student
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+
+    parent_permission = ''
+
+    case gets.chomp.downcase
+    when 'y'
+      parent_permission = true
+
+    when 'n'
+      parent_permission = false
+    else
+      puts('please enter [Y/N]: ')
+    end
+
+    new_student = Student.new(age, name, parent_permission)
+    @people << new_student
+    puts @people
+    run
+
+    puts('Student person created successfully!')
+  end
+
+  # create teacher
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    new_teacher = Teacher.new(specialization, age, name)
+    @people << new_teacher
+    puts('Teacher person created successfully!')
+    run
+  end
+
+  # create book
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+
+    puts 'Author:'
+    author = gets.chomp
+
+    new_book = books.new(title, author)
+    @book << new_book
+
```

Comment:
Good job working on the change requested by the previous reviewer here, but there are a few changes you should fix to make this codebase work as expected.

- [ ] I noticed on line 151, you used lowercase `books.new()` when it was supposed to be in uppercased `Books.new()`. It's a minute change and when I made it uppercased Books, I was able to create a book. Kindly make a fix to that so one can be able to create a book 👍